sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -1698,15 +1700,22 @@ ProgramManager::link(const std::vector<device_image_plain> &DeviceImages,
   for (const device &Dev : Devs)
     PIDevices.push_back(getSyclObjImpl(Dev)->getHandleRef());
 
+  std::string LinkOptionsStr;
+  for (const device_image_plain &DeviceImage : DeviceImages) {
+    const std::shared_ptr<device_image_impl> &InputImpl =
+        getSyclObjImpl(DeviceImage);
+    std::string CompileOpts;
+    applyOptionsFromImage(CompileOpts, LinkOptionsStr,
```

Comment:
Minor.
Can we customize `applyOptionsFromImage` to avoid filling `CompileOpts` when not requested? I think we could, split the function to `applyCompileOptionsFromImage` and `applyLinkOptionsFromImage` which would handle compile and link options separately.

Reply:
I think we could, because I also find it more comfortable. I'll implement it.